lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -344,7 +363,7 @@ struct DifferentiationInvoker {
     return value.adFuncInst;
   }
 
-  std::pair<ApplyInst *, DifferentiationTask *>
+  std::tuple<ApplyInst *, SILFunction *, SILDifferentiableAttr *>
```

Comment:
The triple for indirect differentiation is ``(original `apply` instruction, original function, `[differentiable]` attribute)`` for the parent invoker - the original function and `[differentiable]` attribute can be used to index into `ADContext::invokers` to get the parent invoker.

Your comment made me realize storing the original function is redundant, because the original function can be simply retrieved from the `apply` instruction. Does storing a `std::pair<ApplyInst *, SILDifferentiableAttr *>` (for now) make sense to you?

Reply:
FWIW, `SILDifferentiableAttr` actually stores the original function.